sympy/functions/elementary/trigonometric.py

```diff
@@ -2251,6 +2251,16 @@ def _eval_as_leading_term(self, x):
         else:
             return self.func(arg)
 
+    def _eval_is_real(self):
+        x = self.args[0]
+        if x.is_real == False:
+            return False
+        y = abs(x) - 1
```

Comment:
I suspect that this is connected with the possibility of x being zero in which case asec(0) is not defined. What happens if you add  nonzero=True?